api/geth_backend.go

```diff
@@ -240,6 +240,24 @@ func (b *GethStatusBackend) GetAccounts() ([]multiaccounts.Account, error) {
 	return b.multiaccountsDB.GetAccounts()
 }
 
+func (b *GethStatusBackend) AcceptTerms() error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	if b.multiaccountsDB == nil {
+		return errors.New("accounts db wasn't initialized")
+	}
+
+	accounts, err := b.multiaccountsDB.GetAccounts()
+	if err != nil {
+		return err
+	}
+	if len(accounts) == 0 {
+		return errors.New("accounts is empty")
+	}
+
+	return b.multiaccountsDB.UpdateHasAcceptedTerms(accounts[0].KeyUID, true)
```

Comment:
Is it always `accounts[0]`? 🤔 
UPD: I see now your comment about `HasAcceptedTerms will be set to true when the first account is created.`. So it's the idea, we only store it for the first account? Basically, this setting is "per device"?

Reply:
Correct @igor-sirotin, this setting is per device.

Reply:
But does it make sense to store it in `accounts` table then? 
Maybe we could use some single-row table, not to confuse us?

Reply:
Also, just thought of this now, maybe it should be not a boolean flag, but a `version`? 

So that any time in the future the policy changes and we need to show it again to the user, we could compare the new version with the version agreed by user.

Reply:
> Maybe we could use some single-row table, not to confuse us?

I pondered a lot about this, and in retrospect I wish I went this route in the beginning. The solution using the accounts table was something Andrea and I agreed on was good enough, but yes, there's some confusion around this solution. After the release we can make this improvement, I can personally do it 🙏🏼 

> So that any time in the future the policy changes and we need to show it again to the user, we could compare the new version with the version agreed by user.

@igor-sirotin I thought about this before, but there are many subtle details that were too much for me to solve until Monday (new tentative deadline). We don't have a proper _version_ of the terms to refer to (and we can't use the software version as fallback). The version of a policy would need at least a special major and minor version because not all changes are legally significant to require the user to re-accept terms.

There's a case to be made as well that we don't need to optimize this whole process too much because the policies should change as little as possible or almost never ideally. Legally significant changes in the terms of use & privacy policy are risky. See now, how we will need to make public announcements to help users understand the changes. This is expensive on many layers.

Another detail is that there's no central source of truth for the terms of use and privacy policy. Mobile points to a file, desktop points to another file, legal uses another Google Docs.

With this PR's solution, in a future release, where the terms change (in a legally significant way), we will need to add a quick migration that resets all accounts `hasAcceptedTerms` to false. This will automatically be picked up by the client (mobile only at the moment).

But with all that said, I still think your suggestion is ideal. We just need to outline all the details in the process before jumping in the code.

Reply:
just to clarify, maybe there was some misunderstanding, I suggested to check if there is more than one account and terms have not been accepted that we should show the modal, but I didn't suggest to keep this value within an account row, keeping it in a global separate table is better. I think actually that could be buggy, because accounts can be deleted, and then the user would be asked to reaccept the terms most likely (but haven't checked deeply the code to see if that's the case). 
Either ways I am ok with both in the interest of time of course

Reply:
> With this PR's solution, in a future release, where the terms change (in a legally significant way), we will need to add a quick migration that resets all accounts hasAcceptedTerms to false. This will automatically be picked up by the client (mobile only at the moment).

@ilmotta True, this works as well and is seems much simpler, considering that we don't expect the policy to change frequently.

> Either ways I am ok with both in the interest of time of course

Agree 👍 No need to change anything in release branch, if we're sure that it will not be difficult to change later.

Reply:
> I think actually that could be buggy, because accounts can be deleted, and then the user would be asked to reaccept the terms most likely

This is already how the app works in develop @cammellos. Deleting all accounts brings the user back to the intro screen, where they need to accept terms again. Whether this is good or bad depends a bit on perspective, not necessarily buggy.

So this PR's implementation had this benefit of matching how the app currently works without further changes.

> Agree 👍 No need to change anything in release branch, if we're sure that it will not be difficult to change later.

@igor-sirotin it should be doable to change later.

Reply:
@ilmotta I was referring to maybe a different case, not sure if that happens, but just by looking at the code it might:

1) user has 3 accounts on v1
2) user is presented with a pop up when upgrading
3) user accepts, setting account 1 has `accepted`
4) user deletes account 1

In this case, would the user be presented with terms again?

(Not that is an issue as it's an edge case, but good to be aware if it does happen)

Also my assumption is that the user is presented with terms in case of upgrade only once before login, not after (i haven't checked).

It would be solved easily by marking all accounts as accepted if we want to keep the per account flag.

Apologies if i misunderstood the code, admiteddly i haven t looked at the frontend pr yet :)

Reply:
No no, apologies myself. This is a possibility indeed, and I hope to do the best implementation after the release.